src/cljs_time/internal/parse.cljs

```diff
@@ -285,20 +291,39 @@
           (throw (err))
           out)))))
 
-(defn compile [class {:keys [default-year] :as fmt} values]
-  (let [{:keys [years months days
-                hours HOURS minutes seconds millis
-                meridiem timezone]
-         :as date-map} (->> values
-                            (remove (comp #{:quoted} first))
-                            (into {}))
-        year (.getYear (Date.))
+(defn infer-years
+  [years default-year]
+  (let [year (.getYear (Date.))
         pivot (- year 30)
         century (- year (mod year 100))
         years (or years default-year 0)
         years (cond-> years
                 (< years (mod (+ pivot 50) 100))
-                (+ century))
+                (+ century))]
+    years))
+
+(defn week-date->gregorian
+  [{:keys [weekyear weekyear-week day-of-week] :as date-map}]
```

Comment:
I'm a bit concerned about if the case where someone uses a formatter that includes, E.G., `yyyy` and `xxxxww`.

It's kind of a user-error, but would probably lead them to many WTF!?s when the weekyear is overriding year.

Do you think we should do some validation? There is already a date validation fn at [1], maybe that should be extended to check that both year & weekyear are not set at the same time?

[1]
https://github.com/andrewmcveigh/cljs-time/blob/d31ccf1a656db6d45caa8c3c3fb355de9185dc79/src/cljs_time/internal/core.cljs#L38

Reply:
Will think a bit about what combinations are valid, but yes, adding to `valid-date?` is probably right.

Reply:
Added to `validate-date?`, where you can mix between `years months days` and `weekyears weekyear-weeks day-of-week`. This catches the error a bit later than creating the `xxxx-yyyy-ww` formatter, but I'm not sure whether an `yyyy-yy-yyyy` formatter is valid either?

Reply:
Yeah, you're right actually. There isn't much formatter validation. And this would belong in formatter validation. (rather than data validation)

It's better here than nowhere at least.